services/ui-src/src/utils/forms/dynamicItemFields.ts

```diff
@@ -0,0 +1,114 @@
+import { AnyObject, FormJson } from "types";
+
+// first attempt to combine ilos and plans functions
+
+export const generateDrawerItemFields = (
+  form: FormJson,
+  items: AnyObject[],
+  itemType: string
+) => {
+  const fields = form.fields[0];
+  return {
+    ...form,
+    fields: [
+      {
+        ...form.fields[0],
+        props: {
+          ...form.fields[0].props,
+          choices: [
+            ...updatedItemChoiceList(
+              fields.props?.choices,
+              availableItems(items),
+              itemType
+            ),
+          ],
+        },
+      },
+    ],
+  };
+};
+
+export const parentFieldName = (itemType: string) => {
+  if (itemType === "plan") {
+    return "analysis_applicable";
+  } else {
+    return "plan_ilosUtilizationByPlan";
+  }
+};
+
+const availableItems = (items: AnyObject[]) => {
+  const updatedItemChoices: AnyObject[] = [];
+  items.forEach((item) => {
+    updatedItemChoices.push({
+      ...item,
+      label: item.name,
+      checked: false,
+      children: [
+        {
+          id: `${parentFieldName}_${item.id}`,
+          type: "number",
+          validation: {
+            type: "number",
+            nested: true,
+            parentFieldName: `${parentFieldName}`,
+            parentOptionId: item.id,
+          },
+          props: {
+            decimalPlacesToRoundTo: 0,
+          },
+        },
+      ],
```

Comment:
the analysis methods plan option does not have this nested number field

Reply:
I'm gonna need to link up with you about this cause I'm confused!